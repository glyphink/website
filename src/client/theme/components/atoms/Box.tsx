/**
 * @file Components - Box
 * @module website/theme/components/atoms/Box
 */

import type { Props } from '@glyphink/website/theme'
import {
  forwardRef,
  type ForwardedRef,
  type ForwardRefExoticComponent,
  type JSX
} from 'react'
import type BoxProps from './Box.props.mts'

/**
 * A `<div>` element.
 *
 * - https://developer.mozilla.org/docs/Web/API/HTMLDivElement
 *
 * @see {@linkcode BoxProps}
 * @see {@linkcode ForwardRefExoticComponent}
 *
 * @const {ForwardRefExoticComponent<BoxProps>} Box
 */
const Box: ForwardRefExoticComponent<BoxProps> = forwardRef<
  HTMLDivElement,
  BoxProps
>(
  /**
   * @param {Props<BoxProps>} props
   *  Component props
   * @param {ForwardedRef<HTMLDivElement>} ref
   *  Forwarded ref
   * @return {JSX.Element}
   *  JSX `<div>` element
   */
  (props: Props<BoxProps>, ref: ForwardedRef<HTMLDivElement>): JSX.Element => {
    return <div {...props} ref={ref} />
  }
)

Box.displayName = 'Box'

export default Box
