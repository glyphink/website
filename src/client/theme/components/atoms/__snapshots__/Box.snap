// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:theme/components/atoms/Box > should render <div> 1`] = `<div />`;
